### ISSUE TITLE
Support ESLint Flat config

### ISSUE THREAD

**Issue by sdavids** [testing-library/eslint-plugin-jest-dom#385]
- `eslint-plugin-jest-dom` version: 5.5.0
- `node` version: 22.13.0
- `npm` version: 10.9.2

eslint.config.js
```javascript
import jestDom from 'eslint-plugin-jest-dom';

export default [
  {
    files: ['**/*.test.js'],
    ...jestDom.configs['flat/recommended'],
  },
];
```

Flat config has been the default for 10 months now.

[ESLint 8 has reached EOL](https://eslint.org/blog/2024/09/eslint-v8-eol-version-support/)

While you are at it, you might want to update `README.md` with flat config examples as well.

**Comment by sdavids** [comment]
[Related issue](https://github.com/testing-library/eslint-plugin-testing-library/issues/900)

**Comment by G-Rath** [comment]
@sdavids can you provide more details to what you're actually running into? We already support ESLint v9 + flat config, and I personally have been using it in `cjs` and `mjs` configs without any issues

**Comment by sdavids** [comment]
PEBCAK

`files: ['src/**/*.test.js']` does not match `src/a.test.js`

Should have used `npx eslint --inspect-config` 😕
